pkg/controller/pod.go

```diff
@@ -469,22 +494,62 @@ func (c *Controller) handleDeletePod(key string) error {
 		}
 	}
 
-	if err := c.ovnClient.DeleteLogicalSwitchPort(ovs.PodNameToPortName(name, namespace)); err != nil {
-		klog.Errorf("failed to delete lsp %s, %v", ovs.PodNameToPortName(name, namespace), err)
+	ports, err := c.getPodPorts(name, namespace)
+	if err != nil {
+		klog.Errorf("failed to get attachPorts of pod '%s', %v", name, err)
 		return err
 	}
 
-	if err := c.config.KubeOvnClient.KubeovnV1().IPs().Delete(context.Background(), ovs.PodNameToPortName(name, namespace), metav1.DeleteOptions{}); err != nil {
-		if !k8serrors.IsNotFound(err) {
-			klog.Errorf("failed to delete ip %s, %v", ovs.PodNameToPortName(name, namespace), err)
+	for _, portName := range ports {
+		if err := c.ovnClient.DeleteLogicalSwitchPort(portName); err != nil {
+			klog.Errorf("failed to delete lsp %s, %v", portName, err)
 			return err
 		}
+		if err := c.config.KubeOvnClient.KubeovnV1().IPs().Delete(context.Background(), portName, metav1.DeleteOptions{}); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				klog.Errorf("failed to delete ip %s, %v", portName, err)
+				return err
+			}
+		}
 	}
-
 	c.ipam.ReleaseAddressByPod(key)
 	return nil
 }
 
+func (c *Controller) getPodPorts(pod, namespace string) (ports []string, err error) {
+	defaultPort := ovs.PodNameToPortName(pod, namespace, util.OvnProvider)
```

Comment:
we can set lsp external_ids with the pod namespace/name, so we can more clearly filter the lsp here

Reply:
Rright, I will improve according to the suggestion